src/lib/operators/export_binary.cpp

```diff
@@ -255,17 +255,130 @@ void ExportBinary::_write_segment(const RunLengthSegment<T>& run_length_segment,
   export_values(ofstream, *run_length_segment.end_positions());
 }
 
-void ExportBinary::_export_attribute_vector(std::ofstream& ofstream, const CompressedVectorType type,
-                                            const BaseCompressedVector& attribute_vector) {
+template <typename T>
+void ExportBinary::_write_segment(const FrameOfReferenceSegment<T>& frame_of_reference_segment,
+                                  std::ofstream& ofstream) {
+  Fail("Frame of Reference Segments not implemented for data type");
+}
+
+template <>
+void ExportBinary::_write_segment(const FrameOfReferenceSegment<int32_t>& frame_of_reference_segment,
+                                  std::ofstream& ofstream) {
+  export_value(ofstream, EncodingType::FrameOfReference);
+
+  const auto offset_value_vector_width = [&]() {
```

Comment:
Why do we need this a second time?

Reply:
unified it in one _compressed_vector_width() method